src/node_mgmt/commands.py

```diff
@@ -55,26 +56,23 @@ def env_scan(node):
     scan_result = scanner.do_scan()
     logger.info('Completed environment scan. Submitting results to API and MQTT')
     node.api.post_env_scan(scan_result)
-    if node.mqtt_client.publish(scan_result, topic=MQTT_STATE_TOPIC):
-        logger.info(f"ENV_SCAN [mqtt]: Successfully pushed")
-    else:
-        # For some reason the env_state wasn't pushed successfully
-        logger.warning(f"ENV_SCAN [mqtt]: Push failed")
 
 
 def trigger_config_generation(node, tank_dimensions=None):
-    logger.info('Starting environment scan')
-    scanner = EnvScanner()
-    scan_result = scanner.do_scan()
-    scan_result[GENERATE_NEW_CONFIG_FLAG] = True
+    logger.info('Starting autoconfig trigger')
+    last_env_scan = None
```

Comment:
I don't think there's a need to initialize here.

Reply:
I was afraid it will fail with the error "referenced before assignment". Will remove and test it again

Reply:
The only way that `last_env_scan` would remain undefined is if an exception is raised during instantiation of `KVCache()` on the next line, in which case the code would crash anyway.